src/compiler/utilities.ts

```diff
@@ -3289,6 +3289,10 @@ namespace ts {
         return startsWith(symbol.escapedName as string, "__@");
     }
 
+    export function isPrivateIdentifierSymbol(symbol: Symbol): boolean {

```

Comment:
hmm. I had to figure this out recently and I thought the Right Way was `symbol.valueDeclaration?.kind === SyntaxKind.PrivateIdentifier` or something similar. But I guess this way works too.

Reply:
I would readily believe, or at least hope, that those are equivalent, but this implementation gets to the core of what went wrong in the linked crash. You cannot ask for a name starting with `__#` in `checker.tryGetInModuleExportsAndProperties` and expect to get the private identifier symbol it came from back out, because the checker will underscore-escape that string and try to find a property whose _actual name_ starts with `__#`.